regression/goto-instrument/nondet_static/test.desc

```diff
@@ -0,0 +1,11 @@
+CORE
+main.c
+--nondet-static --show-goto-functions
+^EXIT=0$
+^SIGNAL=0$
+^VERIFICATION SUCCESSFUL$
+g_B = \{ 10, 20, 30 \}
+--
+!(g_B = NONDET\(const signed int \[3l\]\);)
```

Comment:
Is this pattern correct - in particular the leading `!`? I've not written a negative test like this yet so I'm not totally sure of the syntax here. Also, I'd almost be tempted to wildcard match the contents of the NONDET in this pattern e.g. `NONDET\(.*\);` or whatever, since the key thing this patch fixes and we want to preserve is that we don't expect to NONDET the constant array. By making the pattern too specific, it could be possible that a bug might introduce a new way of NONDET'ing this that has slightly different values, which then wouldn't trigger this test to fail - by making the NONDET use a broader regex, this test will then catch any bug that causes a NONDET to appear.

Reply:
Agreed that there's no place for the `!` here.